downloadTemplate skips creating a folder, causes generateSite to fail

**Describe the bug**
When I install `dtcw`, download the template and try to build a microsite, the build fails with an exception (see *below*)
Can be fixed manually with 
````
cd src/docs
mkdir arc42
mv .asciidoctorconfig.adoc arc42/
mv arc42.adoc arc42/
mv chapters arc42/
````
but `downloadTemplate` should do it right in the first place.

*below:*
````
dtcw 0.50 - ##DTCW_GIT_HASH##
docToolchain 3.0.2
OS/arch: Linux/x86_64
Available docToolchain environments: local docker
Environments with docToolchain [3.0.2]: docker
Using environment: docker

Bash is running on WSL which might cause problems with plantUML.
See https://doctoolchain.org/docToolchain/v2.0.x/010_manual/50_Frequently_asked_Questions.html#wsl for more details.

To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/8.1.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 

FAILURE: Build failed with an exception.

* Where:
Script '/home/dtcuser/docToolchain/scripts/AsciiDocBasics.gradle' line: 287

* What went wrong:
A problem occurred evaluating script.
>
                  The file /project/src/docs/arc42/arc42.adoc in HTML config does not exist!
                  Please check the configuration 'inputFiles' in docToolchainConfig.groovy.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 20s
````
**To Reproduce**
Steps to reproduce the behavior:
in an empty folder execute
***WSL (ubuntu 20.04)***
````
curl -Lo dtcw https://doctoolchain.org/dtcw
chmod +x dtcw
./dtcw tasks --group=doctoolchain #enter 'y'
./dtcw downloadTemplate           #enter '1', 'EN', 'withhelp', 'n'
./dtcw generateSite
````
or
***Powershell 5.1***
````
Invoke-WebRequest doctoolchain.org/dtcw.ps1 -Outfile dtcw.ps1
.\dtcw.ps1 tasks --group=doctoolchain
.\dtcw.ps1 downloadTemplate
.\dtcw.ps1 generateSite
````

**Expected behavior**
The microsite is built.

**Configuration**
 - dtcw 0.52
 - docToolchain version: 3.0.2
 - OS: Windows
 - environment: docker


**Additional context**
File system after: ./dtcw downloadTemplate
````
.
├── docToolchainConfig.groovy
├── dtcw
└── src
    ├── docs
    │   ├── .asciidoctorconfig.adoc
    │   ├── arc42.adoc
    │   └── chapters
    │       ├── 01_introduction_and_goals.adoc
    │       ├── 02_architecture_constraints.adoc
    │       ├── 03_system_scope_and_context.adoc
    │       ├── 04_solution_strategy.adoc
    │       ├── 05_building_block_view.adoc
    │       ├── 06_runtime_view.adoc
    │       ├── 07_deployment_view.adoc
    │       ├── 08_concepts.adoc
    │       ├── 09_architecture_decisions.adoc
    │       ├── 10_quality_requirements.adoc
    │       ├── 11_technical_risks.adoc
    │       ├── 12_glossary.adoc
    │       ├── about-arc42.adoc
    │       └── config.adoc
    └── images
        ├── 01_2_iso-25010-topics-EN.drawio.png
        ├── 05_building_blocks-EN.png
        ├── 08-Crosscutting-Concepts-Structure-EN.png
        └── arc42-logo.png
````

--- a/changelog.adoc
+++ b/changelog.adoc
@@ -9,6 +9,8 @@ and this project tries to adhere to https://semver.org/spec/v2.0.0.html[Semantic
 
 === fixes
 
+* https://github.com/docToolchain/docToolchain/pull/1262[#1262: downloadTemplate skips creating a folder, causes generateSite to fail]
+
 === added
 
 === changed

--- a/scripts/downloadTemplate.gradle
+++ b/scripts/downloadTemplate.gradle
@@ -39,8 +39,6 @@ task downloadTemplate(
             template = templates.keySet()[(ant.template as Integer)-1]
         }
         println "${color 'green', 'Install '+template+' documentation template.'}"
-        def outputDir = new File(docDir, inputPath)
-        outputDir.mkdirs()
         if (template in ['arc42', 'req42']) {
             if (template=='arc42') {
                 println "For more information about arc42 see https://arc42.org"
@@ -54,6 +52,8 @@ task downloadTemplate(
             help = ant.help
             antora = ant.antora.toBoolean()
         }
+        def outputDir = new File(docDir, Paths.get(inputPath, template).toString())
+        outputDir.mkdirs()
         download.run {
             src templates[template]
             dest new File(outputDir, 'template.zip')
@@ -170,11 +170,11 @@ ifndef::optimize-content[]
                 configFile.text
                     .replaceAll('//useAntoraIntegration = false', 'useAntoraIntegration = true')
                     .replaceAll('[, \\t\\r\\n]+/[*]{2} inputFiles [*]{2}/',
-                        ",\n\t[file: 'modules/ROOT/pages/index.adoc', formats: ['html','pdf']],\n\t/** inputFiles **/")
+                        ",\n\t[file: '${template}/modules/ROOT/pages/index.adoc', formats: ['html','pdf']],\n\t/** inputFiles **/")
                     .replaceAll('/[*]{2} imageDirs [*]{2}/',
-                        "'modules/ROOT/assets/images/.',\n\t/** imageDirs **/")
+                        "'${template}/modules/ROOT/assets/images/.',\n\t/** imageDirs **/")
                     .replaceAll('[, \\t\\r\\n]+/[*]{2} imageDirs [*]{2}/',
-                        ",\n\t'modules/ROOT/assets/images/.',\n\t/** imageDirs **/")
+                        ",\n\t'${template}/modules/ROOT/assets/images/.',\n\t/** imageDirs **/")
                     .replaceAll("\\[,", "["),
                 'utf-8'
             )

